packages/api/src/controllers/access-control.ts

```diff
@@ -8,4 +14,66 @@ const app = Router();
 accessControl.use("/signing-key", signingKeyApp);
 app.use("/access-control", accessControl);
 
+accessControl.post(
+  "/gate",
+  authorizer({ anyAdmin: true }),
+  validatePost("access-control-gate-payload"),
+  async (req, res) => {
+    const playbackId = req.body.stream.replace("video+", "");
+    const content =
+      (await db.stream.getByPlaybackId(playbackId)) ||
+      (await db.asset.getByPlaybackId(playbackId));
+
+    if (!content) {
+      throw new ForbiddenError("Content not found");
+    }
+
+    if (content.deleted) {
+      throw new ForbiddenError("Stream is deleted");
+    }
+
+    const playbackPolicy = content.playbackPolicy;
+
+    if (playbackPolicy) {
+      if (playbackPolicy.type === "public") {
```

Comment:
**nit:** Could create a variable for the policy type here that already defaults to whatever we want as the default if it's not set. Then IMO the code paths would be clearer, without a 3rd code path a dozen lines below for when the policy is not set. WDYT? Sth like
```suggestion
    const playPolicyType = content.playbackPolicy?.type ?? "public";
    if (playPolicyType === "public") {
       ...
    } else if (playPolicyType === "signed") {
       ...
    } else {
       throw new BadRequestError(`unknown policy type: ${playPolicyType}`)
    }
```

The if/else chain could also be a switch case if you like those (sometimes I do xD)

Reply:
Also note that the final `else` (which would be the `switch` `default` case) is absolutely necessary so we don't allow playback of any playback policy that the current version does not understand. 

So even if you don't refactor into the temp variable I suggested, which is an optional change, you must add explicit handling of an unknown value here, instead of just falling back to allowing playback which is happening below if `playbackPolicy.type` doesn't match any of the cases here.

Reply:
Makes sense! I was using a switch case first, but the check on the string was failing as it was using `strict` and I wasn't able to get true on `playbackPolicy.type === "signed"` :eyes: 

Meanwhile I did it with the if else, putting the final else for unknown policy types as you suggested!

Reply:
Cool! Yeah that switch issue is very weird, IDK what's up. An if/else chain looks great and sometimes better than switch anyway which have several idiosyncracies (like that "temporal something" we had the other day)